src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnection.cs

```diff
@@ -13,15 +13,17 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.NamedPipes.Internal;
 
 internal sealed class NamedPipeConnection : TransportConnection, IConnectionNamedPipeFeature
 {
+    private static readonly ConnectionAbortedException SendGracefullyCompletedException = new ConnectionAbortedException("The named pipe transport's send loop completed gracefully.");
```

Comment:
is there any risk of this shared exception causing problems? in particular I'm thinking of anyone touching `.Data` - also, it isn't *thrown* here, so `.StackTrace` is going to be blank

Reply:
The same exception isn't thrown, it's given to `PipeReader.Complete` and `PipeWriter.Complete`. They capture its details with `ExceptionDispatchInfo.Capture`. If it ends up being rethrown, then it is a copy.

Reply:
👍